src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/ItemsRepeater.cs

```diff
@@ -845,12 +869,27 @@ void OnLayoutChanged(Layout oldValue, Layout newValue)
 			if (newValue != null)
 			{
 				newValue.InitializeForContext(GetLayoutContext());
-				newValue.MeasureInvalidated += InvalidateMeasureForLayout;
-				newValue.ArrangeInvalidated += InvalidateArrangeForLayout;
+				// Uno specific: Using weak subscription..
+				// ItemsRepeater uses a singleton StackLayout.
+				// Subscribing to MeasureInvalidated and failing to unsubscribe is a large memory leak.
+				// Unsubscribing for that in Unloaded failed, because subscription can happen before Loaded and in cases where Loaded/Unloaded are never raised.
+				// The fact that we subscribe in such case feels like a lifecycle bug related to applying templates (things are
+				// initiated in NavigationView.OnApplyTemplate, but probably that OnApplyTemplate call shouldn't have happened).
+				// For now, the only feasible solution is to have a weak event subscription.
+				// NOTE that at the time of writing this, there is another bad subscription that happens early in ItemsRepeater constructor, which isn't the case on WinUI.
+				// However, fixing that bad subscription will still leak due to the lifecycle issue (at least, at the time of writing this).
+
```

Comment:
This is quite wrong. The root cause isn't a lifecycle bug.

It's the fact that the leaking ItemsRepeater is in a Flyout in NavigationView template. So, it won't get loaded without opening the flyout.

The proper fix is replicating what Fluent v2 styles do, i.e, not use the singleton `StackLayout` for this specific `ItemsRepeater`.